Change mouse cursors while joystick scrolling

JoppyFurr opened the pull request:
This patch adds mouse cursors to the joystick scrolling feature.

On Tiberian Sun, which itself had joystick scrolling, the same cursors are
used as in the original game.

As the older games did not have joystick scrolling, I've had to make a
best-effort guess as to what were suitable cursors.
  -> When scrolling in all directions is available, the default arrow
     cursor is used.
  -> When a direction is blocked, the edge-scrolling blocked direction
     cursor is used.

deniz1a commented:
I don't see any cursors?

JoppyFurr commented:
deniz1a: In Tiberian Sun, Joystick scrolling should use the same cursors as the original. Green four-way arrows during right-click-drag, with the arrows turning red when you reach the limit of a direction.

In the older games, which don't have the same icon set, I've used the edge-scrolling-blocked cursors when reaching the scroll limits of a direction. When all directions are available I've used the normal mouse pointer, as the older games don't seem to have anything that looks like the four-way arrow cursor in TS.

deniz1a commented:
Oh ok I thought this also added arrows when all directions are available in red alert. There could be a custom image created from four edge arrows to make something like the TS one.

pchote commented:
This blinks the scroll cursor whenever you issue an order with our default (right-click) settings.  That will need to be fixed before merging.

JoppyFurr committed:
Change mouse cursors while joystick scrolling

This patch adds mouse cursors to the joystick scrolling feature.

On Tiberian Sun, which itself had joystick scrolling, the same cursors are
used as in the original game.

As the older games did not have joystick scrolling, I've had to make a
best-effort guess as to what were suitable cursors.
  -> When scrolling in all directions is available, the default arrow
     cursor is used.
  -> When a direction is blocked, the edge-scrolling blocked direction
     cursor is used.

JoppyFurr commented:
Hi,

I've just made a change to fix up the cursor blink on right-click.


Cheers,
Joppy

abcdefg30 commented:
Looks good to me. :+1: / :white_check_mark:

Mailaender commented:
Works as promised. :white_check_mark: Nice polish followup of https://github.com/OpenRA/OpenRA/pull/9518.

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/2253cc95eb70037f241edf9e0d1113a079d52ad7...64de10fc1a27ea8380c402108753b8c559c89701)

Mailaender commented:
Attention @Phrohdoh this will break your RA2 mod if you don't manually update the cursor sequences.

Mailaender commented:
@Rampastring Can you we use the Dawn of the Tiberium Age joystick scroll cursors in OpenRA?

Rampastring commented:
@Mailaender Yes, as long as you credit Dawn of the Tiberium Age in some way.
